compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala

```diff
@@ -27,17 +27,22 @@ object ClasspathFromClassloader {
             classpathBuff ++=
               cl.getURLs.iterator.map(url => Paths.get(url.toURI).toAbsolutePath.toString)
           case _ =>
-            // HACK: We can't just collect the classpath from arbitrary parent
-            // classloaders since the current classloader might intentionally
-            // filter loading classes from its parent (for example
-            // BootFilteredLoader in the sbt launcher does this and we really
-            // don't want to include the scala-library that sbt depends on
-            // here), but we do need to look at the parent of the REPL
-            // classloader, so we special case it. We can't do this using a type
-            // test since the REPL classloader class itself is normally loaded
-            // with a different classloader.
-            if (cl.getClass.getName == classOf[AbstractFileClassLoader].getName)
+            if cl.getClass.getName == classOf[AbstractFileClassLoader].getName then
+              // HACK: We can't just collect the classpath from arbitrary parent
+              // classloaders since the current classloader might intentionally
+              // filter loading classes from its parent (for example
+              // BootFilteredLoader in the sbt launcher does this and we really
+              // don't want to include the scala-library that sbt depends on
+              // here), but we do need to look at the parent of the REPL
+              // classloader, so we special case it. We can't do this using a type
+              // test since the REPL classloader class itself is normally loaded
+              // with a different classloader.
               collectClassLoaderPaths(cl.getParent)
+            else if cl.getClass.getName.contains("AppClassLoader") then
```

Comment:
can you check `cl == cl.getSystemClassLoader` instead?

Reply:
or Maybe cl.getClass == cl.getSystemClassLoader.getClass, not sure.

Reply:
Maybe `cl.getParent eq cl.getSystemClassLoader`. I will check.

Reply:
No, the parent is the `PlatformClassLoader`. It works with `cl eq ClassLoader.getSystemClassLoader`